deploy/complete/terraform/provider.tf

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at http://oss.oracle.com/licenses/upl.
+# 
+
+terraform {
+  required_version = ">= 0.12.16"
+  # required_providers {
+  #   kubernetes = ">= 1.8"
+  #   helm = ">= 1.1.1"
+  #   oci = ">= 3.73"
+  # }
+}
+
+provider "oci" {
+  tenancy_ocid     = var.tenancy_ocid
+  region           = var.region
+}
+
+provider "kubernetes" {
+  load_config_file = "false"
+  cluster_ca_certificate = base64decode(yamldecode(data.oci_containerengine_cluster_kube_config.oke_cluster_kube_config.content)["clusters"][0]["cluster"]["certificate-authority-data"])
+  host = yamldecode(data.oci_containerengine_cluster_kube_config.oke_cluster_kube_config.content)["clusters"][0]["cluster"]["server"]
+  exec {
```

Comment:
Add a comment reference to 
https://docs.cloud.oracle.com/en-us/iaas/Content/ContEng/Tasks/contengdownloadkubeconfigfile.htm#notes ?
Might help users understand whats going on here

Reply:
This does not download the kubeconfig. This documentation will confuse.

The resource "local_file" "kubeconfig" is the one that generate the kubeconfig, and just used if some use the `terraform apply`. And just in case is needed. For convenience, not needed for the scripts.

Reply:
Not about download - a description as to why we have the `exec` block here

Reply:
Also format this file too while you’re at it